### PR TITLE
create container with clang

### PR DESCRIPTION
**Pull request by ObiWahn**
- add clang container
- provide entrypoint that runs scan-build

### CODE REVIEW

**Comment by ObiWahn** [comment]
https://ufile.io/y9awb3m3 - sample - expires in 30 days

**Comment by ObiWahn** [comment]
https://ufile.io/6uel2pp3 - more hits

File: containers/clang.docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        curl \
+        tar \
+        git \
+        rsync \
+        parallel \
+        tree \
+        jq \
+        && apt-get clean
+
+## install packages for python
```
Comment (fceller) [inline containers/clang.docker/Dockerfile]: @ObiWahn can you put this in one apt-get?

Do we need both python and python3?

File: containers/clang.docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+RUN mkdir "$ARANGO_INSTALL" && chmod 777 "$ARANGO_INSTALL"
+
+## other setup - ditaa, lockfile, bash_lib
+COPY setup /setup
+RUN /setup
+
+WORKDIR "$ARANGO_INSTALL"
+
+## setup entrypoint
+COPY entrypoint /entrypoint
```
Comment (fceller) [inline containers/clang.docker/Dockerfile]: @ObiWahn can you rename this to entrypoint.sh

File: containers/clang.docker/entrypoint [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+export ARANGO_BUILD="${ARANGO_BUILD_CLANG:-"$ARANGO_WORK/clang-build"}"
+
+cd "$ARANGO_SOURCE" || { echo "could not change in arangodb source directroy"; exit 1; }
+
+#TODO check .git/config instead
+[[ -f VERSION ]]  ||  { echo "please make sure you are in a valid checkout"; exit 1; }
+[[ -f VERSIONS ]] ||  { echo "please make sure you are in a valid checkout"; exit 1; }
+
+echo "calling build clang script"
+"$ARANGO_OSKAR/containers/clang.docker/clang_build" "$@" 2>&1 | tee "$ARANGO_WORK/clang-build.log"
```
Comment (fceller) [inline containers/clang.docker/entrypoint]: @ObiWahn can you rename this clang_build.sh

File: containers/clang.docker/setup [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
Comment (fceller) [inline containers/clang.docker/setup]: @ObiWahn please rename to setup.sh

File: scripts/clangScanBuild [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
Comment (fceller) [inline scripts/clangScanBuild]: @ObiWahn can you rename this to clang-scan-build.sh

**Comment by ObiWahn** [comment]
I would like to keep the files without suffix. Especially because `.sh` suggests posix-shell compatible what is not the case. Here a bit more on the topic that is frequently given by bash devs to new users: http://www.talisman.org/~erlkonig/documents/commandname-extensions-considered-harmful/

File: containers/clang.docker/Dockerfile (hunk 1 above)
Comment (ObiWahn) [reply containers/clang.docker/Dockerfile]: Python2 is about to expire ( https://pythonclock.org/ ). Unfortunately our build still requires this versions. I think we should remove the python2 package as soon as possible.

Comment (dothebart) [reply containers/clang.docker/Dockerfile]: fix it ;-P https://github.com/refack/GYP3

**Comment by fceller** [comment]
> @fceller I would like to keep the files without suffix. Especially because `.sh` suggests posix-shell compatible what is not the case. Here a bit more on the topic that is frequently given by bash devs to new users: http://www.talisman.org/~erlkonig/documents/commandname-extensions-considered-harmful/

yes, but in all other projects we use ".sh". If they are now without ".sh" it will just create confusion.

File: containers/clang.docker/Dockerfile (hunk 1 above)
Comment (ObiWahn) [reply containers/clang.docker/Dockerfile]: It will not be fixed here. But if we have to change we will not need to build a new container...

**Comment by ObiWahn** [comment]
https://ufile.io/3sn0ten9 - paths excluded

File: containers/clang.docker/Dockerfile (hunk 1 above)
Comment (ObiWahn) [reply containers/clang.docker/Dockerfile]: @dothebart please do not bring up the 2% increase in container size :)

**Comment by ObiWahn** [comment]
Closed as there is not sufficient interest.

**Comment by joerg84** [comment]
@fceller Was there missing interest here?

**Pull request closed without merging**